sport/app/cricket/jobs/CricketStatsJob.scala

```diff
@@ -42,9 +42,12 @@ class CricketStatsJob(paFeed: PaFeed) extends Logging {
         Days.daysBetween(cricketMatch.gameDate.toLocalDate, LocalDate.now).getDays > 5
       ).map(_.matchId).toSeq
 
-      paFeed.getMatchIds(team).map { matchIds =>
+      val matchIds = if (includeHistorical) paFeed.getHistoricalMatchIds(team) else paFeed.getCurrentMatchIds(team)
 
-        val matches = matchIds.diff(loadedMatches).take(10)
+      matchIds.map { matchIds =>
+
+        // never fetch more than 10 matches
```

Comment:
Careful!

There are still county cricket matches happening during the world cup so you might find you get to 10 before you fetch today's world cup match, for example. If you order by time and limit to the 10 most recent that would be fine.

Reply:
Good spot. In this case we get away with it as we're fetchig the last 10 matches just for the 10 teams in the world cup. I think I need to bump it to 12 though I guess the finalists will play 11 matches